assembl/static2/js/app/components/debate/brightMirror/sentimentMenu.jsx

```diff
@@ -0,0 +1,90 @@
+// @flow
+import React from 'react';
+import get from 'lodash/get';
+
+import { ApolloClient } from 'react-apollo';
+import { OverlayTrigger } from 'react-bootstrap';
+import { Translate } from 'react-redux-i18n';
+
+import { MEDIUM_SCREEN_WIDTH, SENTIMENT_TOP_OFFSET, SENTIMENT_RIGHT_OFFSET } from '../../../constants';
+import Sentiments from '../common/sentiments';
+import getSentimentStats from '../common/sentimentStats';
+import fictionSentimentDefinitions from './fictionSentimentDefinition';
+
+export type Props = {
+  position: { x: number, y: number },
+  postId: string,
+  sentimentCounts: ?SentimentCountsFragment,
+  mySentiment: ?string,
+  client: ApolloClient,
+  screenWidth: number,
+  refetchPost: Function,
+  isPhaseCompleted: boolean
+};
+
+const SentimentMenu = ({
+  postId,
+  mySentiment,
+  client,
+  screenWidth,
+  refetchPost,
+  position,
+  sentimentCounts,
+  isPhaseCompleted
+}: Props) => {
+  let count = 0;
+  const totalSentimentsCount = sentimentCounts
+    ? sentimentCounts.like + sentimentCounts.disagree + sentimentCounts.dontUnderstand + sentimentCounts.moreInfo
+    : 0;
+  return (
+    <div
+      className="sentiment-container hidden-xs hidden-sm hidden-md"
+      style={{
+        top: `${position ? position.y + SENTIMENT_TOP_OFFSET : 0}px`,
+        left: `${position ? position.x / 2 - SENTIMENT_RIGHT_OFFSET : 0}px`
+      }}
+    >
+      <Sentiments
+        sentimentCounts={sentimentCounts}
+        mySentiment={mySentiment}
+        placement="right"
+        client={client}
+        postId={postId}
+        isPhaseCompleted={isPhaseCompleted}
+        onCompleted={refetchPost}
+        customSentimentDefinitions={fictionSentimentDefinitions}
+      />
+      {totalSentimentsCount > 0 ? (
+        <OverlayTrigger overlay={getSentimentStats(totalSentimentsCount, sentimentCounts, mySentiment)} placement="right">
+          <div className="sentiments-count margin-m">
+            <div>
+              {fictionSentimentDefinitions.reduce((result, sentiment) => {
```

Comment:
Maybe extract this block in a render function? just to keep jsx neat

Reply:
Map doesn't fit for the usage as we are going through all items in fictionSentimentDefinitions but we might not want a result with same size as input

Reply:
It s better to not extract the function as we need ```count``` to be modified in this function